### ISSUE TITLE
Settle on a USB communications library

### ISSUE THREAD

**Issue by qdot** [libnifalcon/libnifalcon#42]
At the time libnifalcon was written, it was hard to find a usb/ftdi library that supported our async needs. That's probably changed in the past few years. Would be nice to settle on a single cross platform library that suits our requirements.

**Comment by agirault** [comment]
Are you talking about libusb1 for Unix vs libftd2xx for Windows?
Would be nice to have libusb1 for all systems. VRPN that I use to interface with the Falcon through LibNiFalcon uses libusb1 : https://github.com/vrpn/vrpn/blob/master/cmake/FindLibusb1.cmake

**Comment by FabienDanieau** [comment]
I agree, libusb seems to be a good option. If it works well with Windows now we won't need FTD2XX drivers anymore.

**Comment by agirault** [comment]
I have been testing with libusb1 on windows, managed to fix any linker issues etc. See #43 

`setFalconComm<FalconCommLibUSB>()` is called instead of `setFalconComm<FalconCommFTD2XX>()` in the `FalconDevice` constructor.

Calling `getDeviceCount()` works successfully, but `open()` fails, and I get the following errors:
- `falconDevice->getErrorCode()` : 2000
- `falconDevice->getFalconComm->getDeviceErrorCode()` : -12

**Comment by agirault** [comment]
from @mcuee on https://github.com/libusb/libusb/issues/185#issuecomment-218641145 :

> In order to use libusb under Windows, you have to use a supported driver --
> WinUSB for example and not FTDI vendor driver.

Yout toughts on that @qdot ?

**Comment by qdot** [comment]
OH. Now I remember why I did it this way. :|

I kept the FTDI driver in on windows for people that want to use software that requires either the Novint SDK OR libnifalcon.

Would be nice to keep both around. That said, the Novint SDK has a chance of disappearing/not working at any point these days, since it hasn't been updated in years, so being able to support libusb on windows would be nice. We can just create a Zadig installer for WinUSB drivers.

I guess the problem at that point is, given we'll have a platform will support both drivers, how do we have programs figure out which driver to use on windows?
